contrib/download_cmake.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# Script to download newest version of cmake on linux (or mac)
+# saves you the trouble of compiling it if you don't have root
+set -e # stop on failure
+cd "$(dirname "$0")"/.. # run in top-level directory
```

Comment:
It seems messy to me, but that's an aesthetic choice.  This doesn't seem too conceptually different to me than downloading source and compiling the binary to use it during build time, like we do with e.g. `patchelf`, but this isn't a strong objection.  At the very least, we should add `.gitignore` rules for wherever this does end up.